widgetssdk/src/main/java/com/glia/widgets/GliaWidgets.java

```diff
@@ -184,7 +184,11 @@ public synchronized static EngagementLauncher getEngagementLauncher(@Nullable Li
      */
     @NonNull
     public synchronized static EntryWidget getEntryWidget(@Nullable List<String> queueIds) {
-        setupQueueIds(queueIds);
+        if (Dependencies.glia().isInitialized()) {
+            setupQueueIds(queueIds);
+        } else {
+            Logger.w(TAG, "Attempt to get EntryWidget before SDK initialization");
```

Comment:
But actually what is the purpose of letting integrators to get the entrywidget without SDK initialization?

Reply:
As per requirements:
> Integrators can launch the Entry Widget view after the SDK setup. If SDK hasn’t been initialized or was initialized with an error - Widgets SDK should not crash, it should show an [Entry Widget with an error state](https://www.figma.com/design/x5BLxm2UWpm3TuerdqmYi8/Glia-Mobile-App-Pages?node-id=11753-4938&node-type=SECTION&t=csKDTOMlup7L17Hn-0).

Initially, I suggested throwing an exception and crashing the app so that this misconfiguration issue was discovered early during development time.

I remember @gugalo brought up the idea to avoid crashing at the kick-off meeting and it has been approved.

@DavDo What alternatives do you have in mind?

Reply:
> Initially, I suggested throwing an exception and crashing the app so that this misconfiguration issue was discovered early during development time.

I'm still up for this suggestion because this initialization part and errors thrown during misconfiguration must be disruptive to show that misconfiguration during development.
I think if it goes to the production, that is the integrator's problem.

However, if we decide to go with this solution, I think the only way is to still configure queues and log errors.
```java
public synchronized static EntryWidget getEntryWidget(@Nullable List<String> queueIds) {
        Dependencies.getConfigurationManager().setQueueIds(queueIds);

        if (!Dependencies.glia().isInitialized()) {
            Logger.w(TAG, "Attempt to get EntryWidget before SDK initialization");
        }

        return Dependencies.getEntryWidget();
    }
```

Reply:
Could we upgrade the log to error?
I think we want to get alerted if integrators are attempting to use entry widgets before SDK is init

Reply:
I've made changes as per the code that you suggested.
Also, raised this question in Slack.